emoji-ui/src/components/emojiItem.js

```diff
@@ -0,0 +1,25 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import './emojiItem.css';
+
+class EmojiItem extends Component {
+  static propTypes = {
+    isSelected: PropTypes.bool.isRequired,
+    url: PropTypes.string.isRequired,
+    clickHandler: PropTypes.func.isRequired.isRequired,
+  };
+
+  render() {
+    const { isSelected, url, clickHandler } = this.props;
+    const styles = { backgroundImage: `url(${url})` };
+    const containerClasses = `Emoji-container${isSelected ? ' Emoji-container-selected' : ''}`;
+
+    return (
+      <div className={containerClasses}>
+        <button style={styles} className='Emoji-item' onClick={clickHandler}></button>
```

Comment:
it may make sense to pass the name to the clickhandler inside this component. but that can be left until you develop how this is used.

Reply:
ohhh! yes, i will pass the name in. i forgot we need the name to extract.